packages/edit-site/src/components/global-styles/global-styles-provider.js

```diff
@@ -45,22 +45,35 @@ const cleanEmptyObject = ( object ) => {
 };
 
 function useGlobalStylesUserConfig() {
-	const { globalStylesId, settings, styles } = useSelect( ( select ) => {
-		const _globalStylesId =
-			select( coreStore ).__experimentalGetCurrentGlobalStylesId();
-		const record = _globalStylesId
-			? select( coreStore ).getEditedEntityRecord(
-					'root',
-					'globalStyles',
-					_globalStylesId
-			  )
-			: undefined;
```

Comment:
Can we instead rely on the "isResolving state" of both these two resolvers to mark it ready (instead of checking active themes).

Reply:
I also had that idea; I will test how it works in practice.

Reply:
Update the logic to use resolvers [69f0c34](https://github.com/WordPress/gutenberg/pull/46073/commits/69f0c34c220ceaa499bcc551150de40b28b234e0).